File: WorkIt/app/src/main/java/prot3ct/workit/data/remote/base/UserDataContract.java
package prot3ct.workit.data.remote.base;

import io.reactivex.Observable;
import prot3ct.workit.view_models.ProfileDetailsViewModel;

public interface UserDataContract {
    Observable<ProfileDetailsViewModel> getProfileDetails(int userId);

    Observable<Boolean> updateProfile(String fullName, String phone, String profilePictureAsString);
}

File: WorkIt/app/src/main/java/prot3ct/workit/data/remote/base/UserDataContract.kt
package prot3ct.workit.data.remote.base

import io.reactivex.Observable
import prot3ct.workit.view_models.ProfileDetailsViewModel

interface UserDataContract {

    fun getProfileDetails(userId: Int): Observable<ProfileDetailsViewModel>

    fun updateProfile(
        fullName: String,
        phone: String,
        profilePictureAsString: String?
    ): Observable<Boolean>
}